commons/com.b2international.index/src/com/b2international/index/revision/DefaultRevisionIndex.java

```diff
@@ -142,27 +142,27 @@ public RevisionCompare compare(final String branch) {
 	
 	@Override
 	public RevisionCompare compare(final String branch, final int limit) {
-		return compare(getBaseRef(branch), getBranchRef(branch), limit);
+		return compare(getBaseRef(branch), getBranchRef(branch), limit, true);
 	}
 	
 	@Override
 	public RevisionCompare compare(final String baseBranch, final String compareBranch) {
-		return compare(baseBranch, compareBranch, COMPARE_DEFAULT_LIMIT);
+		return compare(baseBranch, compareBranch, COMPARE_DEFAULT_LIMIT, true);
 	}
 	
 	@Override
-	public RevisionCompare compare(final String baseBranch, final String compareBranch, final int limit) {
-		return compare(getBranchRef(baseBranch), getBranchRef(compareBranch), limit);
+	public RevisionCompare compare(final String baseBranch, final String compareBranch, final int limit, boolean preserveComponentChanges) {
```

Comment:
I think a better configuration option would be the opposite, `excludeComponentChanges`. The default functionality is to always include them, so allowing optional exclusion of them is the functionality we need. Please rename.

Reply:
Alternatively, one could define the set of Operation literal to exclude.